From 27fe3b0ed353f78dc0a0fa4ce1be600901bb8c90 Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Mon, 24 Apr 2023 12:55:00 +0530
Subject: [PATCH] Add Home Key for MultiUser MultiDisplay CarFeature.

Added Home Button for passenger user display.

Tracked-On: OAM-108949
Signed-off-by: Ankit Agrawal <ankit.agarwal@intel.com>
---
 .../res/drawable/ic_home_key.xml              | 26 +++++++++++++++++++
 .../res/layout/navigation_keys.xml            |  9 ++++++-
 .../launcher/LauncherActivity.java            | 15 +++++++++++
 3 files changed, 49 insertions(+), 1 deletion(-)
 create mode 100644 tests/MultiDisplaySecondaryHomeTestLauncher/res/drawable/ic_home_key.xml

diff --git a/tests/MultiDisplaySecondaryHomeTestLauncher/res/drawable/ic_home_key.xml b/tests/MultiDisplaySecondaryHomeTestLauncher/res/drawable/ic_home_key.xml
new file mode 100644
index 000000000..077b26e7d
--- /dev/null
+++ b/tests/MultiDisplaySecondaryHomeTestLauncher/res/drawable/ic_home_key.xml
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+     Copyright (C) 2018 The Android Open Source Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="28dp"
+    android:height="28dp"
+    android:viewportWidth="28"
+    android:viewportHeight="28">
+
+    <path
+        android:fillColor="@color/gray"
+        android:pathData="M 14 7 C 17.8659932488 7 21 10.1340067512 21 14 C 21 17.8659932488 17.8659932488 21 14 21 C 10.1340067512 21 7 17.8659932488 7 14 C 7 10.1340067512 10.1340067512 7 14 7 Z" />
+</vector>
diff --git a/tests/MultiDisplaySecondaryHomeTestLauncher/res/layout/navigation_keys.xml b/tests/MultiDisplaySecondaryHomeTestLauncher/res/layout/navigation_keys.xml
index 0f9a720c8..208300380 100644
--- a/tests/MultiDisplaySecondaryHomeTestLauncher/res/layout/navigation_keys.xml
+++ b/tests/MultiDisplaySecondaryHomeTestLauncher/res/layout/navigation_keys.xml
@@ -8,5 +8,12 @@
         android:id="@+id/back_button"
         android:layout_width="wrap_content"
 	android:layout_height="wrap_content"
-        android:background="@drawable/ic_back_key"/>
+	android:background="@drawable/ic_back_key"/>
+
+     <Button
+         android:id="@+id/home_button"
+         android:layout_width="wrap_content"
+         android:layout_height="wrap_content"
+         android:background="@drawable/ic_home_key"/>
+
 </LinearLayout>
diff --git a/tests/MultiDisplaySecondaryHomeTestLauncher/src/com/android/car/multidisplay/launcher/LauncherActivity.java b/tests/MultiDisplaySecondaryHomeTestLauncher/src/com/android/car/multidisplay/launcher/LauncherActivity.java
index 4712b5968..bcd900497 100644
--- a/tests/MultiDisplaySecondaryHomeTestLauncher/src/com/android/car/multidisplay/launcher/LauncherActivity.java
+++ b/tests/MultiDisplaySecondaryHomeTestLauncher/src/com/android/car/multidisplay/launcher/LauncherActivity.java
@@ -23,8 +23,10 @@ import android.animation.AnimatorListenerAdapter;
 import android.app.ActivityOptions;
 import android.app.AlertDialog;
 import android.app.Application;
+import android.app.Service;
 import android.car.Car;
 import android.car.CarOccupantZoneManager;
+import android.content.ComponentName;
 import android.content.Context;
 import android.content.Intent;
 import android.content.SharedPreferences;
@@ -34,6 +36,7 @@ import android.hardware.display.DisplayManager;
 import android.hardware.input.InputManager;
 import android.os.Bundle;
 import android.os.SystemClock;
+import android.os.UserHandle;
 import android.util.Log;
 import android.view.Display;
 import android.view.Gravity;
@@ -98,6 +101,7 @@ public class LauncherActivity extends FragmentActivity implements AppPickedCallb
     private int mDisplayId;
     private LinearLayout mOverlay;
     private Button mBackKey;
+    private Button mHomeKey;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
@@ -217,6 +221,7 @@ public class LauncherActivity extends FragmentActivity implements AppPickedCallb
         LayoutInflater inflater = (LayoutInflater) getSystemService(LAYOUT_INFLATER_SERVICE);
         mOverlay = (LinearLayout) inflater.inflate(R.layout.navigation_keys, null);
         mBackKey = mOverlay.findViewById(R.id.back_button);
+        mHomeKey = mOverlay.findViewById(R.id.home_button);
 
         WindowManager wm = (WindowManager) this.getSystemService(WINDOW_SERVICE);
         wm.addView(mOverlay, params);
@@ -228,6 +233,16 @@ public class LauncherActivity extends FragmentActivity implements AppPickedCallb
             }
         });
 
+        mHomeKey.setOnClickListener(new View.OnClickListener() {
+            @Override
+             public void onClick(View v) {
+                 Intent appIntent = new Intent();
+                 appIntent.setComponent(new ComponentName("com.android.car.multidisplay", "com.android.car.multidisplay.launcher.LauncherActivity"));
+                 appIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                 getApplicationContext().startActivityAsUser(appIntent, android.os.Process.myUserHandle());
+             }
+        });
+
     }
 
     void injectInputEvent(int keycode) {
-- 
2.17.1

